src/tests/ftest/util/test_utils_pool.py

```diff
@@ -10,21 +10,91 @@
 import json
 
 from test_utils_base import TestDaosApiBase, LabelGenerator
-from avocado import fail_on
+from avocado import fail_on, TestFail
 from command_utils import BasicParameter
 from exception_utils import CommandFailure
 from pydaos.raw import (DaosApiError, DaosPool, c_uuid_to_str, daos_cref)
 from general_utils import check_pool_files, DaosTestError
 from server_utils_base import ServerFailed, AutosizeCancel
 from dmg_utils import DmgCommand
 
+POOL_NAMESPACE = "/run/pool/*"
+
+
+def add_pool(test, namespace=POOL_NAMESPACE, create=True, connect=True, index=0, **params):
+    """Add a new TestPool object to the test.
+
+    Args:
+        test (Test): the test to which the pool will be added
+        namespace (str, optional): TestPool parameters path in the test yaml file. Defaults to
+            POOL_NAMESPACE.
+        create (bool, optional): should the pool be created. Defaults to True.
+        connect (bool, optional): should the pool be connected. Defaults to True.
+        index (int, optional): Server index for dmg command. Defaults to 0.
+
+    Returns:
+        TestPool: the new pool object
+
+    """
+    pool = TestPool(
+        namespace=namespace, context=test.context, dmg_command=test.get_dmg_command(index),
+        label_generator=test.label_generator)
+    pool.get_params(test)
+    if params:
+        pool.update_params(**params)
+    if create:
+        pool.create()
+    if create and connect:
+        pool.connect()
+
+    # Add a step to remove this pool when the test completes and ensure their is enough time for the
+    # pool destroy to be attempted - accounting for a possible dmg command timeout
+    test.increment_timeout(200)
+    test.register_cleanup(remove_pool, test=test, pool=pool)
+
+    return pool
+
+
+def remove_pool(test, pool):
```

Comment:
The timeout that `add_pool()` is incrementing is the one avocado uses for the process that runs the test `tearDown()` method when the original test timeout expires before running `tearDown()`.  Avocado does not provide a mechanism for the `remove_pool()` method to decrement the timeout of the process that is running the `remove_pool()` method.

Reply:
OK. Figured that might be the case, but just wanted to raise the concern. I'm a little worried that this change might overall improve CI stability at the expense of throughput. Maybe as a follow-on it might make sense to figure out how we can somehow know when one pool's teardown is taking longer than the per-pool timeout increment.